.github/workflows/tests@v1.yml

```diff
@@ -28,9 +28,33 @@ jobs:
           java-version: ${{ matrix.java-version }}
           distribution: 'adopt'
 
+      - name: Get POM hash
+        id: get-pom-hash
+        run: echo "value=${{ hashFiles('**/pom.xml') }}" >> "$GITHUB_OUTPUT"
+
+      - uses: actions/cache/restore@v4
+        id: java-cache
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-${{ matrix.java-version }}-maven-${{ steps.get-pom-hash.outputs.value }}
+
       - name: Compile source and tests
         run: mvn -B compile test-compile -Dfmt.skip=true -Dclirr.skip=true -Danimal.sniffer.skip=true
 
+      - name: Download test dependencies
```

Comment:
Wouldnt dependencies be downloaded in previous step during `compile` and `test-compile`?

Reply:
There are maven dependencies that are not being downloaded.

Reply:
Right, I forgot for example about runtime dependencies